drivers/dac/ltc2672/ltc2672.c

```diff
@@ -0,0 +1,526 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <errno.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include "ltc2672.h"
+#include "no_os_spi.h"
+#include "no_os_error.h"
+#include "no_os_util.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/******************************************************************************/
+
+/**
+ * @brief Device and communication init function
+ * @param device - ltc2672 descriptor to be initialized
+ * @param init_param - Initial parameter for descriptor
+ * @return 0 in case of success, errno errors otherwise
+ */
+int ltc2672_init(struct ltc2672_dev **device,
+		 struct ltc2672_init_param *init_param)
+{
+	int ret;
+	struct ltc2672_dev *descriptor;
+
+	descriptor = (struct ltc2672_dev *)calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+
+	ret = no_os_spi_init(&descriptor->comm_desc, &init_param->spi_init);
+	if (ret)
+		goto error_init;
+
+	descriptor->active_device = init_param->active_device;
+	descriptor->prev_command = 0x00000000;
+	descriptor->global_toggle = false; // default toggle is false
+	descriptor->neg_current = init_param->neg_current;
+
+	/* Default at power up: ALL channels turned off */
+	for (int i = 0; i < LTC2672_TOTAL_CHANNELS; i++) {
+		descriptor->out_spans[i] = LTC2672_OFF;
+		descriptor->max_currents[i] = LTC2672_OFF;
+	}
+
+	*device = descriptor;
+
+	return 0;
+
+error_init:
+	free(descriptor);
+	return ret;
+}
+
+/**
+ * @brief Remove resources allocated by the init function
+ * @param device  - ltc2672 descriptor
+ * @return 0 in case of success, errno errors otherwise
+ */
+int ltc2672_remove(struct ltc2672_dev *device)
+{
+	int ret;
+
+	if (!device)
+		return -ENODEV;
+
+	ret = no_os_spi_remove(device->comm_desc);
+
+	free(device);
+
+	return ret;
+}
+
+/**
+ * @brief Write raw register value
+ * @param device - ltc2672 descriptor
+ * @param comm - command byte to send
+ * @param is_32 - flag if command is 32 bit (True) or 24 bit (False) format
+ * @return 0 in case of success, negative error code otherwise
+ */
+int ltc2672_transaction(struct ltc2672_dev *device, uint32_t comm, bool is_32)
+{
+	int ret;
+	uint8_t raw_array[4]; // highest bytes transmitted is 4 bytes
+	uint16_t bytes;
+	uint32_t read_back = 0;
+
+	if (is_32)
+		bytes = 4;
+	else
+		bytes = 3;
+
+	for (int i = bytes; i > 0; i--)
```

Comment:
you have `bytes` declared above as uint16_t, why declare another one ?
also, this inline declaration is tricky, i don't think it works with all C compilers so best to drop it